src/create-glamorous.js

```diff
@@ -161,6 +159,45 @@ function createGlamorous(splitProps) {
       )
       return GlamorousComponent
     }
+
+    /**
+     * This returns a glamorousComponentFactory that will be composed with
+     * the provided properties.
+     *
+     * @param {...Object|Function} outerProps the properties to provide to the
+     *  GlamorousComponent.
+     *  If any of these are functions, they will be invoked with the component
+     *  props and the return value is used.
+     * @return {Function} the glamorousComponentFactory function.
+     */
+    glamorousComponentFactory.withProps = (...outerProps) => {
+      // This is a higher order function of `glamorousComponentFactory`
+      return (...styles) => {
```

Comment:
Let's use a function declaration here too. That way this function can have a name which will be used in stack traces 😀